openslides_backend/action/actions/mediafile/upload.py

```diff
@@ -93,30 +88,24 @@ def update_organization_instance(self, instance: Dict[str, Any]) -> Dict[str, An
         instance["filesize"] = len(decoded_file)
         id_ = instance["id"]
         mimetype_ = instance["mimetype"]
-        self.media.upload_mediafile(file_, id_, mimetype_)
-        return instance
-
-    def update_meeting_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
-        instance["create_timestamp"] = round(time())
-        instance["mimetype"] = mimetypes.guess_type(instance["filename"])[0]
-        if instance["mimetype"] is None:
-            raise ActionException(f"Cannot guess mimetype for {instance['filename']}.")
-        decoded_file = base64.b64decode(instance["file"])
-        instance["filesize"] = len(decoded_file)
         if instance["mimetype"] == "application/pdf":
             instance["pdf_information"] = self.get_pdf_information(decoded_file)
+        self.media.upload_mediafile(file_, id_, mimetype_)
+        collection, _ = self.get_owner_data(instance)
+        if collection == "meeting":
+            instance["filename"] = filename_
+            instance = self.update_access_fields(instance)
```

Comment:
Do this before uploading the mediafile so you save time and storage in case of an error.

Reply:
Moved the media service upload at the end of the update_instance method.